python/pyarrow/tests/test_fs.py

```diff
@@ -1112,6 +1110,20 @@ def test_s3_options():
     assert pickle.loads(pickle.dumps(fs2)) == fs2
     assert fs2 != fs
 
+    # Check if we have set a C++ AWS Standard retry strategy by testing
+    # if the methods on the strategy quack like a Standard strategy.
+    fs = S3FileSystem(
+        retry_strategy=AwsStandardS3RetryStrategy(max_attempts=5))
+    assert isinstance(fs, S3FileSystem)
+    assert fs.retry_strategy.ShouldRetry(None, 1)
```

Comment:
Hmm... how does this work? `ShouldRetry` is not exposed in Python, is it?

Reply:
In any case, I don't think these tests are really needed. Just test that passing constructor arguments works.